### ISSUE TITLE
mockPrompt does not throw errors when invalid options are passed

### ISSUE THREAD

**Issue by niftylettuce** [saojs/sao#51]
e.g. if we have a prompt of `name` and the following, then it does not throw an error with ava for example (so we can't test code coverage)

```js
  // ...
  prompts: {
    name: {
      message: 'What is the name of the new package',
      default: ':folderName:',
      validate: val => {
        return slug(val) === val
          ? true
          : `Please change the name from "${val}" to "${slug(val)}"`;
      }
    },
    // ...
```

**Comment by egoist** [comment]
Fixed in `v0.21.6`

**Comment by niftylettuce** [comment]
thank you.
